examples/pytorch/deepspeed-demo/pytorch_deepspeed_demo.yaml

```diff
@@ -0,0 +1,38 @@
+apiVersion: "kubeflow.org/v1"
+kind: PyTorchJob
+metadata:
+  name: pytorch-deepspeed-demo
+spec:
+  pytorchReplicaSpecs:
+    Master:
+      replicas: 1
+      restartPolicy: OnFailure
+      template:
+        spec:
+          containers:
+            - name: pytorch
+              image: kubeflow/pytorch-deepspeed-demo:latest
+              command:
+                - torchrun
```

Comment:
Do we need to set the `--nnodes` and `--nproc_per_node` parameters for `torchrun` ?

Reply:
Yes, I didn't set them during my testing because there was only one GPU on the node. I have added them now.

Reply:
@Syulin7 No, actually, you no need to set the parameters for `torchrun`, setting the correct environment related parameters(or using env in the operator case) is the responsibility of the operator.

If you set them, the parameters will overwrite the env which will work with no doubt, but we don't encourage our use to use it this way. Since we use operator, we leave those staff to the operator.

Reply:
@andreyvelich I'm not aware of any deprecation of ENV variables usage from the pytorch part.
In the link it mentioned that `--use-env` is deprecated for `torchrun`, it means the usage of args like `torchrun --use-env` is invalid anymore. Anyway, the cli argument and environment are both valid way and **identical** to set environ related information to the pytorch framework.

For operator, I think the operator should take care of all that kind of staff for the user, that's one reason why people should use the operator. Otherwise, people can use Job or even Statefulset or some other kind of interface to run pods in kubernetes, then launch the distributed job with argument with torchrun. 

So, in my opinion, to run a pytorch distributed job one can do it simply with `torchrun train.py`, the operator make it happened.

Reply:
Make sense @kuizhiqing.
I see, I found this how the use the `PET_` env variables with `torchrun`
https://github.com/pytorch/pytorch/blob/main/torch/distributed/argparse_util.py#L44-L46
Do you know if they have any PyTorch docs which explains it ?

Reply:
@andreyvelich Yes, you are right. I don't think they have docs specifically to explain the env usage since the usage of arguments and envs are processed equally.

Using env is the right way for operator to interact with program, and it was the original design of operator, c.f. https://github.com/kubeflow/training-operator/pull/1573.

Well, it's indeed a little bit confusion for user to understand the meaning of all the env, especially when it comes to PET_ prefix and switch with other frameworks, c.f. https://github.com/kubeflow/training-operator/pull/1840.

One more thing is worth mention that, the argument can not be change dynamically, while env can.

Reply:
That makes sense.
@Syulin7 In that case, let's rely on Training Operator to setup those `torchrun` arguments based on the configuration.

Reply:
> So, in my opinion, to run a pytorch distributed job one can do it simply with torchrun train.py, the operator make it happened.

@kuizhiqing @andreyvelich Agree, we should rely on Training Operator to setup envs.